### PR TITLE
[Graphbolt] Fix asseration bug

### PR DESCRIPTION
**Pull request by peizhou001**
## Description
Fix bug in `sample_neighbors` assert for fanout length.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/graph_storage/csc_sampling_graph.py [hunk 1]
```diff
@@ -280,9 +280,11 @@ def sample_neighbors(
...
             greater than or equal to 0."
         if self.metadata and self.metadata.edge_type_to_id:
-            assert len(self.metadata.edge_type_to_id) == fanouts.size(
-                0
-            ), "Fanouts should have the same number of elements as etypes."
+            assert len(self.metadata.edge_type_to_id) in [
+                fanouts.size(0),
+                1,
+            ], "Fanouts should have the same number of elements as etypes or \
+                should have a length of 1."
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/graph_storage/csc_sampling_graph.py:287]: to handle case `fanout=-1`?

Comment (peizhou001) [reply python/dgl/graphbolt/graph_storage/csc_sampling_graph.py:287]: No, the length of `fanout` could be the length of edge types or **1**, **1** was missed before.

Comment (Rhett-Ying) [reply python/dgl/graphbolt/graph_storage/csc_sampling_graph.py:287]: do we have dedicated testcase for `1`

Comment (peizhou001) [reply python/dgl/graphbolt/graph_storage/csc_sampling_graph.py:287]: Yes, in sample neighbors

Comment (Rhett-Ying) [reply python/dgl/graphbolt/graph_storage/csc_sampling_graph.py:287]: as test already in latest code, CI is throwing assertion exception?

Comment (peizhou001) [reply python/dgl/graphbolt/graph_storage/csc_sampling_graph.py:287]: No, the tested graph doesn't have metadata.

Comment (Rhett-Ying) [reply python/dgl/graphbolt/graph_storage/csc_sampling_graph.py:287]: then we could add one which has metadata and access the `1` code path?

Comment (peizhou001) [reply python/dgl/graphbolt/graph_storage/csc_sampling_graph.py:287]: sure , let me add one

**Review by Rhett-Ying — APPROVED** [review]

**Comment by peizhou001** [comment]
@dgl-bot

**Pull request merged**